resolwe_bio/processes/import_data/sra_file.py

```diff
@@ -0,0 +1,403 @@
+        show_advanced = BooleanField(label='Show advanced options', default=False)
+
+        class Advanced:
+            """Advanced options."""
+            min_spot_id = IntegerField(label='Minimum spot ID', required=False)
+            max_spot_id = IntegerField(label='Maximum spot ID', required=False)
+            min_read_len = IntegerField(label='Minimum read length', required=False)
+            clip = BooleanField(label='Clip adapter sequences', default=False)
+            aligned = BooleanField(label='Dump only aligned sequences', default=False)
+            unaligned = BooleanField(label='Dump only unaligned sequences', default=False)
+
+        advanced = GroupField(Advanced, label='Advanced options', hidden='!show_advanced')
+
+    class Output:
+        """Output fields to process ImportSraSingle."""
+        fastq = ListField(FileField(), label='Reads file')
+        fastqc_url = ListField(
+            FileHtmlField(),
+            label='Quality control with FastQC',
+        )
+        fastqc_archive = ListField(FileField(), label='Download FastQC archive')
+
+    def run(self, inputs, outputs):
+        """Run the analysis."""
+        for srr in inputs.sra_accession:
+            cmd = Cmd['fastq-dump']['--gzip']
+            if inputs.advanced.min_spot_id:
+                cmd = cmd['--minSpotId'][inputs.advanced.min_spot_id]
+            if inputs.advanced.max_spot_id:
+                cmd = cmd['--maxSpotId'][inputs.advanced.max_spot_id]
+            if inputs.advanced.min_read_len:
+                cmd = cmd['--minReadLen'][inputs.advanced.min_read_len]
+            if inputs.advanced.clip:
+                cmd = cmd['--clip']
+            if inputs.advanced.aligned:
+                cmd = cmd['--aligned']
+            if inputs.advanced.unaligned:
+                cmd = cmd['--unaligned']
+            return_code, _, _ = cmd[srr] & TEE(retcode=None)
+            if return_code:
+                self.error(f'Download of {srr} reads with fastq-dump failed.')
+
+        fastq_gz = glob.glob('*.fastq.gz')
+        outputs.fastq = fastq_gz
+
+        print('Postprocessing FastQC...')
+        stderr = run_fastqc([fastq_gz], './fastqc')
+        # FastQC writes both progress and errors to stderr and exits with code 0.
+        # Catch if file is empty, wrong format... (Failed to process) or
+        # if file path does not exist, file cannot be read (Skipping).
+        if 'Failed to process' in stderr or 'Skipping' in stderr:
+            self.error('Failed while processing with FastQC.')
+
+        for fastqc_zip in glob.glob('fastqc/*_fastqc.zip'):
+            move(fastqc_zip, '.')
+
+        fastqc_url = []
+        fastqc_dir = Path('fastqc')
+        for report_dir in fastqc_dir.iterdir():
+            if not report_dir.is_dir():
+                continue
+
+            fastqc_url.append({
+                "file": str(report_dir / 'fastqc_report.html'),
```

Comment:
Instead of `str()` you could use `x.name` which could be prettier to read. Maybe.

Reply:
Prettier to read but we want full paths and not only filenames here.